perun-core/src/main/java/cz/metacentrum/perun/core/impl/modules/attributes/urn_perun_resource_attribute_def_def_sshkeysTargetUser.java

```diff
@@ -0,0 +1,49 @@
+package cz.metacentrum.perun.core.impl.modules.attributes;
+
+import cz.metacentrum.perun.core.api.Attribute;
+import cz.metacentrum.perun.core.api.AttributeDefinition;
+import cz.metacentrum.perun.core.api.AttributesManager;
+import cz.metacentrum.perun.core.api.Resource;
+import cz.metacentrum.perun.core.api.exceptions.InternalErrorException;
+import cz.metacentrum.perun.core.api.exceptions.WrongAttributeAssignmentException;
+import cz.metacentrum.perun.core.api.exceptions.WrongAttributeValueException;
+import cz.metacentrum.perun.core.api.exceptions.WrongReferenceAttributeValueException;
+import cz.metacentrum.perun.core.impl.PerunSessionImpl;
+
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+/**
+ * Module for users' sshkeys
+ *
+ * @author Zdenek Strmiska <zdenek.strm@gmail.com>
+ * @date 27.7.2017
+ */
+
+
+public class urn_perun_resource_attribute_def_def_sshkeysTargetUser {
+
+    public void checkAttributeValue(PerunSessionImpl perunSession, Resource resource, Attribute attribute) throws InternalErrorException, WrongAttributeValueException, WrongReferenceAttributeValueException, WrongAttributeAssignmentException {
+        String key = (String) attribute.getValue();
+        if (key == null) {
+            throw new WrongAttributeValueException(attribute, resource, "sshkey can't be empty");
+        }
+
+        Pattern pattern = Pattern.compile("^(?!-)[-_.a-zA-Z0-9]+$");
```

Comment:
Could be done when initializing the class. You don't have to do that each time you have to check an attribute.